src/helpers.coffee

```diff
@@ -233,3 +233,28 @@ exports.nameWhitespaceCharacter = (string) ->
     when '\r' then 'carriage return'
     when '\t' then 'tab'
     else string
+
+exports.dump = (args..., obj) ->
+  util = require 'util'
+  console.log args..., util.inspect obj, no, null
```

Comment:
> We put the dots on the right in the CoffeeScript codebase; but besides, I don’t think we want debugging helpers in the codebase.

@GeoffreyBooth ok updated dots - can already picture having a linting config in the codebase with rules like this...

As far as including this helper, I've been using it constantly on my `prettier` branch while developing, but obviously it's not needed in production. The argument I'd see in favor of including it would be to reference it in the docs for hacking on the compiler ("a convenient way to inspect eg a node object is to use `helpers.dump()`"). But it's a simple enough util to recreate manually when needed and tbh I don't know what's standard practice here

Reply:
So far we haven’t been including debugging helpers, I think probably because everyone has their own opinion as to how they’d want their `dump` function to behave and that just leads to chaos. There is one exception, in that there’s a built-in debugging logger at the top of `rewriter.coffee`, and I confess I had a hand in that one; it used to be just a comment (“uncomment the next line to print the token stream”) and I added the environment variables so that all one had to do to enable the debugging output was set the environment variable rather than uncomment the line, because I was afraid of accidentally committing the uncommented line back into the codebase.

So I can’t really talk 😄but personally I’d prefer to hold the line at the rewriter debugger, and avoid putting a totally generic “print any object” debugging helper in the codebase, especially since it gets included all over the place and it’s an invitation for people to bikeshed and reimplement it according to their own preferences. But my preference doesn’t need to be the final say here, if others feel strongly otherwise.